src/Nethermind/Nethermind.Trie/NibbleExtensions.cs

```diff
@@ -142,6 +145,40 @@ public static byte[] ToBytes(ReadOnlySpan<byte> nibbles)
             return bytes;
         }
 
+        public static byte[] CompactToHexEncode(byte[] compactPath)
+        {
+            if (compactPath.Length == 0)
+            {
+                return compactPath;
+            }
+            int nibblesCount = compactPath.Length * 2 + 1;
+            byte[]? array = null;
+            Span<byte> nibbles = nibblesCount < StackAllocLengthLimit
+                ? stackalloc byte[nibblesCount]
+                : array ??= ArrayPool<byte>.Shared.Rent(nibblesCount);
+
+            try
```

Comment:
Don't need try/finally here, if throw an exception doesn't matter too much if don't return array to the shared array pool